pandera/io.py

```diff
@@ -106,18 +126,22 @@ def _serialize_schema(dataframe_schema):
             for index_stats in statistics["index"]
         ]
 
+    if statistics["checks"] is not None:
+        checks = _serialize_dataframe_stats(statistics["checks"])
+
     return {
         "schema_type": "dataframe",
         "version": __version__,
         "columns": columns,
+        "checks": checks,
         "index": index,
         "coerce": dataframe_schema.coerce,
         "strict": dataframe_schema.strict,
     }
 
 
-def _deserialize_check_stats(check, serialized_check_stats, pandas_dtype):
-    def handle_stat_dtype(stat):
+def _deserialize_check_stats(check, serialized_check_stats, pandas_dtype=None):
+    def handle_stat_dtype(stat, pandas_dtype):
```

Comment:
what's the reason for adding the `pandas_dtype` arg here? the enclosed `handle_stat_dtype` should have access to `pandas_dtype` in the outer function scope.

Reply:
I believe @jeffzi changed this to avoid the nonlocal keyword. If I had to guess, namebinding only works on variables defined in the `_deserialize_check_stats` function. The pandas_dtype is a lookup from "outside" so it doesn't bind. I think I tried leaving out the nonlocal key and it doesn't work.

Reply:
Actually `nonlocal` is not required if we don't modify the outer scope variable. I should have removed the argument when I removed the inference of the dtype from the stat. I just pushed a commit that removes the argument.